dapp: More config options in README

TransmissionsDev opened the pull request.

TransmissionsDev commented:
@d-xo Need some help on some of these env vars. 

I'm fine with just linking to these hevm ones:

https://github.com/dapphub/dapptools/blob/master/src/hevm/README.md#environment-variables

But some of them are unclear even to me, can you possibly add descriptions for each of them?

Other thing I need some help with are some of these env vars:

https://github.com/dapphub/dapptools/blob/a2160ec71e287db2e8d648ae3d057231cf6fb4d5/src/dapp/libexec/dapp/dapp#L154-L170

A lot of them seem to overlap (rpc is mentioned 3x), any chance you could help with descriptions?

d-xo commented:
@TransmissionsDev I think this should be everything. 

p.s. fyi @MrChico `DAPP_TEST_BALANCE_CALLER` seemed to be unused, so I removed it and renamed `DAPP_TEST_BALANCE_CREATE` to `DAPP_TEST_BALANCE`. I would also be fine to add it back, but I didn't really know what it was supposed to do. Do you know?

TransmissionsDev commented:
> @TransmissionsDev This is great, I think I added everything that was missing?
> 
> p.s. fyi @MrChico `DAPP_TEST_BALANCE_CALLER` seemed to be unused, so I removed it and renamed `DAPP_TEST_BALANCE_CREATE` to `DAPP_TEST_BALANCE`. I would also be fine to add it back, but I didn't really know what it was supposed to do. Do you know?

woohoo this is awesome! 

lgtm, moving out of draft
